### DISCUSSION TITLE
User Auth Migration from Basic Auth, Need Advice

### DISCUSSION THREAD

**Discussion by indrazm** [pocketbase/pocketbase#3232] [Q&A]
Hey Everyone.

I am currently building an LMS platform. I am using prisma + postgreSQL.
I wanted to migrate All my user authentication that using basic auth (Email + Hashed Password) to Pocketbase.

Is there any way to do this ? 
Many Thanks.

**Comment by ganigeorgiev** [comment]
Since you have hashed passwords I assume that by "Basic Auth" you don't mean the [Basic Authentication scheme](https://datatracker.ietf.org/doc/html/rfc7617), but a regular email+password authentication.

In this case, what hashing function do you use for the passwords?

Note that you can safely migrate passwords in PocketBase only if they are hashed with bcrypt.
For every other case you'll have to implement a custom auth middleware that on the first user login will take the submitted plain text password, will verify it against the old hash and then will create and save a new bcrypt one (_and eventually delete the old hash_).

But without details about your data structure it will be difficult to generalize.

PocketBase doesn't have builtin mechanisms at the moment to easily import external data and devs are required to define their collections structure upfront either from the Admin UI or programmatically (_in the future there will be table auto discovery that will make the task slightly easier and provide more SQLite "native" dev experience_).

**If your current stack with Prisma and PostgreSQL works for you, then I'd suggest to stick with it.**
While bugs are fixed as soon as they are identified, I can't recommend PocketBase for production critical applications yet, unless you are fine with applying some manual migration steps from time to time (_especially if using as framework/toolkit_) and working with not fully documented apis.

**Comment by indrazm** [comment]
Thank you for your honest advice. 

I guess, i'll just wait till ver 1.0.0.
Many Thanks!

**Comment by rodydavis** [comment]
I am going to attempt a migration with Firebase users that use **scrypt** algorithm!
